.github/workflows/docker-slurm.yml

```diff
@@ -48,6 +48,8 @@ jobs:
         run:  |
           pip install flake8
           flake8 src tests
+          pip install black
+          black src tests
```

Comment:
It will not fail the CI, it just runs black and does the initial check for now. I'm working on adding something in a script that could run to confirm no unformatted code was found from running `black`.

Reply:
Perhaps all that is needed is `black --check --diff`.  As long as that command returns a non-zero exit status if it finds any differences, I think it would work.

Reply:
We have successfully passed the `make lint` CI step and failed the `black` step due to the reformatting of code.

```
Installing collected packages: typing-extensions, pathspec, mypy-extensions, black
Successfully installed black-24.1.1 mypy-extensions-1.0.0 pathspec-0.12.1 typing-extensions-4.9.0
reformatted /home/runner/work/ExascaleWorkflowSandbox/ExascaleWorkflowSandbox/tests/test_parsl_flux_mpi_hello.py

All done! ✨ 🍰 ✨
1 file reformatted, 4 files left unchanged.
Usage: black [OPTIONS] SRC ...
```

The Documentation on the README I believe should be updated to reflect our most recent additions of `black` and `flake8`. Maybe something telling the developer/user "you need to run `black` to format your code before it can be merged and here's the command `python -m black src tests`